layers/core_checks/cc_ray_tracing.cpp

```diff
@@ -215,67 +217,234 @@ bool CoreChecks::ValidateAccelerationStructuresMemoryAlisasing(const LogObjectLi
 
-        const bool other_info_in_update_mode = other_info.mode == VK_BUILD_ACCELERATION_STRUCTURE_MODE_UPDATE_KHR;
+                if (dst_as_state && dst_as_state->VkHandle() != src_as_state->VkHandle()) {
+                    const vvl::range<VkDeviceAddress> dst_as_range = dst_as_state->GetDeviceAddressRange();
 
-        if (other_dst_as_state) {
-            auto other_info_scratches = GetBuffersByAddress(other_info.scratchData.deviceAddress);
-            const VkDeviceSize assumed_other_scratch_size =
-                rt::ComputeScratchSize(rt_build_type, device, other_info, pp_range_infos[other_info_j]);
+                    if (const vvl::range<VkDeviceAddress> dst_as_src_as_intersection = dst_as_range & src_as_range;
+                        dst_as_src_as_intersection.non_empty()) {
+                        skip |=
+                            LogError("VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03668",
+                                     LogObjectList(cmd_buffer, src_as_state->VkHandle(), dst_as_state->VkHandle()),
+                                     info_i_loc.dot(Field::dstAccelerationStructure),
+                                     "overlaps with srcAccelerationStructure, which is in update mode, on device address range %s.",
+                                     string_range_hex(dst_as_src_as_intersection).c_str());
+                    }
+                }
+
+                const AddressRange src_as_address_range = {src_as_range, info_i, AddressRangeOrigin::SrcAccelStruct};
+                const std::optional<AddressRange> overlapped_address_range = insert_address(address_ranges, src_as_address_range);
+                if (overlapped_address_range.has_value()) {
+                    switch (overlapped_address_range->origin) {
+                        case AddressRangeOrigin::SrcAccelStruct: {
+                            // Valid overlap, source acceleration structures being read only
+                            break;
+                        }
+                        case AddressRangeOrigin::DstAccelStruct: {
+                            if (const auto other_dst_as_state = Get<vvl::AccelerationStructureKHR>(
+                                    p_infos[overlapped_address_range->info_i].dstAccelerationStructure)) {
+                                const vvl::range<VkDeviceAddress> src_as_other_dst_as_intersection =
+                                    src_as_address_range.range & overlapped_address_range->range;
+
+                                skip |= LogError(
+                                    "VUID-vkCmdBuildAccelerationStructuresKHR-dstAccelerationStructure-03701",
+                                    LogObjectList(cmd_buffer, src_as_state->VkHandle(), other_dst_as_state->VkHandle()),
+                                    info_i_loc.dot(Field::srcAccelerationStructure),
+                                    ", in update mode, overlaps with dstAccelerationStructure of pInfos[%" PRIu32
```

Comment:
this looks like

> vkCmdBuildAccelerationStructuresKHR(): pInfos[1].srcAccelerationStructure , in update mode, overlaps with dstAccelerationStructure of pInfos[0], on device address range [0x761a654a8000, 0x761a654a8090).

So maybe 

```suggestion
                                    "is in update mode, which overlaps with dstAccelerationStructure of pInfos[%" PRIu32
```

Reply:
the `which` makes you think that what is overlapping is "update mode"
But I can go "is in update mode, **but** overlaps with ..."